### PR TITLE
Add more matrix norms.

### PR DESCRIPTION
**Pull request by argriffing**
SciPy and NumPy already have a few matrix norms accessible through the `ord` argument of `norm`; these include the Frobenius norm, the induced 1-norm and inf-norm, the spectral norm, and three other functions that are not really norms.  This PR expands the selection and provides an interface that is more flexible than overloading a single `ord` argument.

### CODE REVIEW

File: scipy/linalg/_matrix_norms.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+
+from .decomp import _asarray_validated
+from .decomp_svd import svdvals
+
+__all__ = ['elementwise_norm', 'frobenius_norm', 'nuclear_norm',
+           'spectral_norm', 'schatten_norm', 'induced_norm', 'ky_fan_norm']
+
+_npv = NumpyVersion(np.__version__)
+
+_nrm2_prefix = {'f': 's', 'F': 'sc', 'D': 'dz'}
```
Comment (ewmoore) [inline scipy/linalg/_matrix_norms.py]: Doesn't appear to be used anywhere in this file.

**Comment by ewmoore** [comment]
If we want different functions for all of these, I think using a submodule instead of adding them all to the linalg namespace might be better.  One could then call `norm.ky_fan(...)` etc. and We wouldn't have quite so many similar functions at the top level.

I'm also not sure that `elementwise_norm` is the best name.  I don't know that I would expect to find that providing a p-norm.  Is that a standard term that I've just missed?

**Comment by argriffing** [comment]
> I'm also not sure that `elementwise_norm` is the best name. I don't know that I would expect to find that providing a p-norm. Is that a standard term that I've just missed?

I've tried to distinguish the vector p-norm norm of the flattened matrix from induced matrix p-norms and from Schatten matrix p-norms.  Wikipedia calls the vector p-norms of the flattened matrix [entrywise norms](http://en.wikipedia.org/wiki/Matrix_norm#.22Entrywise.22_norms) which I guess I've mistyped in the PR as `elementwise_norm`.

File: scipy/linalg/_matrix_norms.py (hunk 1 above)
Comment (argriffing) [inline scipy/linalg/_matrix_norms.py]: This is now removed.

**Comment by argriffing** [comment]
> If we want different functions for all of these, I think using a submodule instead of adding them all to the linalg namespace might be better. One could then call `norm.ky_fan(...)` etc. and We wouldn't have quite so many similar functions at the top level.

If the functions are wanted and if a submodule is used, then I guess it would have to be called something other than `norm`.  I'd thought about this a bit, but I couldn't come up with anything great.  If the functions are added to the linalg namespace as in the current state of the PR, then I guess a section would need to be added to the linalg [`__init__.py`](https://github.com/scipy/scipy/blob/master/scipy/linalg/__init__.py).

**Comment by argriffing** [comment]
The build (not tests) seems to have timed out.

**Comment by argriffing** [comment]
Closing as too ambitious.

**Comment by ev-br** [comment]
Why closing, I think it is a worthy addition. 
(I don't have intelligent suggestions about the interface)

**Comment by argriffing** [comment]
> Why closing

It seems to be doing too much at once given scipy's mature state of development, and it's causing drag on the other open norm-related PRs.  Maybe it could be re-opened if the backlogs clear a bit.

**Pull request closed without merging**